mcp/sse.go

```diff
@@ -259,7 +260,10 @@ func (s sseServerConn) Read(ctx context.Context) (jsonrpc.Message, error) {
 }
 
 // Write implements jsonrpc2.Writer.
-func (s sseServerConn) Write(ctx context.Context, msg jsonrpc.Message) error {
+func (s *sseServerConn) Write(ctx context.Context, msg jsonrpc.Message) error {
+	s.writeMu.Lock()
```

Comment:
Ditto.

Reply:
D'oh. This made me realize that the write was *already* guarded.